qendpoint-store/src/main/java/com/the_qa_company/qendpoint/compiler/SparqlRepository.java

```diff
@@ -541,6 +519,37 @@ private ClosableResult<?> execute0(RepositoryConnection customConnection, String
 
 			ParsedQuery parsedQuery = QueryParserUtil.parseQuery(QueryLanguage.SPARQL, sparqlQuery, null);
 
+			// Substitute [AUTO_LANGUAGE] with the client's language
+			if (acceptLanguageHeader != null) {
+				List<Locale.LanguageRange> languageRanges = Locale.LanguageRange.parse(acceptLanguageHeader);
+				Locale locale = Locale.lookup(languageRanges, Arrays.asList(Locale.getAvailableLocales()));
+				try {
+					parsedQuery.getTupleExpr().visit(new AbstractQueryModelVisitor<Exception>() {
+						@Override
+						public void meet(Var node) throws Exception {
+							if (node.getValue() instanceof Literal literal) {
+								boolean isStringDatatype = literal.getDatatype() == null
+										|| literal.getDatatype().equals(CoreDatatype.XSD.STRING.getIri());
+								if (isStringDatatype) {
+									String value = literal.getLabel();
+									if (value.equals("[AUTO_LANGUAGE]")) {
+										ValueFactory vf = SimpleValueFactory.getInstance();
+										Literal newLiteral = vf.createLiteral(locale.getLanguage());
```

Comment:
In a same way, we don't want the server language for the client, what we want is the list of languages accepted by the client, for example if my header is:

```
fr,fr-FR;q=0.9,en;q=0.8,en-GB;q=0.7,en-US;q=0.6
```

And I put `"[AUTO_LANGUAGE],ru"`, I would want to have

```
fr,fr-FR,en,en-GB,en-US,ru
```

(so the same without the weights)